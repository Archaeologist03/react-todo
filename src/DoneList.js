import React from 'react';

function DoneList(props) {
    let doneListMapped = props.doneArr.map(listItem => {
        return (
            <li 
                className="listItem"
                key={Math.random()}>
                    <span className="listItemName">{listItem}</span>
                    <a className="listDeleteLink listsBtns" href="#" ><span className="listItemDelete" onClick={() => props.deleteItem(listItem)}>Delete</span></a>
            </li>
        )
    });

    return(
        <ul>
            {doneListMapped}
        </ul>
    ) 

    
}






export default DoneList;
