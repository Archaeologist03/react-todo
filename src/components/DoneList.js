import React from 'react';

function DoneList(props) {
  let doneListMapped = props.doneArr.map(listItem => {
    return (
      <li className="listItem" key={Math.random()}>
        <span className="listItemName">{listItem}</span>
        <a className="listDeleteLink listsBtns" >
          <span
            className="listItemDelete"
            onClick={() => props.onDeleteFromDone(listItem)}>
            Delete
          </span>
        </a>
      </li>
    );
  });

  return <ul>{doneListMapped}</ul>;
}

export default DoneList;
