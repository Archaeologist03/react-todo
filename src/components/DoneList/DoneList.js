import React from 'react';
import { connect } from 'react-redux';

import { deleteFromDone } from '../../store/actions/doneActions';

import '../../scss/listItems/listItem.scss';

function DoneList(props) {
  let doneListMapped = props.doneArr.map(listItem => {
    return listItem.name ? (
      <li className='listItem' key={listItem.id || Math.random()}>
        <span className='listItemName'>{listItem.name}</span>
        <a
          onClick={e => e.preventDefault()}
          className='listDeleteLink listsBtns'
          href='delete'>
          <span
            className='listItemDelete'
            onClick={() => props.onDeleteFromDone(listItem)}>
            Delete
          </span>
        </a>
      </li>
    ) : null;
  });

  return <ul>{doneListMapped}</ul>;
}

const mapStateToProps = state => {
  return {
    doneArr: state.app.done,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onDeleteFromDone: item => dispatch(deleteFromDone(item)),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(DoneList);
