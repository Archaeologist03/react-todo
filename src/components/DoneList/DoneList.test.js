import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { DoneList } from './DoneList';

describe('DoneList', () => {
  it('should render DoneList with given props(task1)', () => {
    const props = {
      doneArr: [{ name: 'task1', id: 1 }],
      onDeleteFromDone: jest.fn(),
    };
    const wrapper = shallow(<DoneList {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render listItemName element with task name as children', () => {
    const props = {
      doneArr: [{ name: 'task1', id: 1 }],
      onDeleteFromList: jest.fn(),
    };
    const wrapper = shallow(<DoneList {...props} />);
    const listItemNameClassElement = wrapper.find('.listItemName');
    expect(listItemNameClassElement.props().children).toBe(
      props.doneArr[0].name,
    );
  });

  it('should call onDeleteFromDone fn from props on click with name arg', () => {
    const props = {
      doneArr: [{ name: 'task1', id: 1 }, { name: 'task1', id: 1 }],
      onDeleteFromDone: jest.fn(),
    };

    const wrapper = shallow(<DoneList {...props} />);
    const listItemDeleteClassElement = wrapper.find('.listItemDelete');

    listItemDeleteClassElement.at(0).simulate('click');
    expect(props.onDeleteFromDone).toHaveBeenCalledWith(props.doneArr[0]);

    listItemDeleteClassElement.at(1).simulate('click');
    expect(props.onDeleteFromDone).toHaveBeenCalledWith(props.doneArr[1]);
  });
});
