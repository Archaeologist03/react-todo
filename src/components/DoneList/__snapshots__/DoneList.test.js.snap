// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoneList should render DoneList with given props(task1) 1`] = `
<ul>
  <li
    className="listItem"
    key="1"
  >
    <span
      className="listItemName"
    >
      task1
    </span>
    <a
      className="listDeleteLink listsBtns"
      href="delete"
      onClick={[Function]}
    >
      <span
        className="listItemDelete"
        onClick={[Function]}
      >
        Delete
      </span>
    </a>
  </li>
</ul>
`;
